src/app/shared/pipes/duration.ts

```diff
@@ -7,3 +7,18 @@ export class DurationToMinPipe implements PipeTransform {
     return duration.minutes();
   }
 }
+
+@Pipe({ name: 'duration' })
+export class DurationToReadable implements PipeTransform {
+  transform(duration: number): string {
+    if (duration < 60) {
+      return `${Math.floor(duration)} min`;
+    } else if (duration < 60 * 24) {
+      return `${Math.floor(duration / 60)} hours`;
+    } else if (duration < 60 * 24 * 30) {
+      return `${Math.floor(duration / 60 / 24)} days`;
+    } else {
+      return `${Math.floor(duration / 60 / 24 / 30)} months`;
+    }
```

Comment:
The duration is in seconds (cfr the spec), not in minutes. That explains why you get results such as "1610 months", it is actually 26 months, which is quite realistic for a task which has been started 2y ago and not completed yet.
For the negative value,s I'll try to see if it can/should be fixed in the db, but anyway, you have to handle it properly. For instance, for the first category, you could just do `${Math.max(Math.floor(duration),0)} min`;

Reply:
Actually, for duration, probably you should re-use what has already been done with the "Duration" class (and the "toMin" pipe).

Reply:
fixed